pipelines/curate_rewards.py

```
@@ -0,0 +1,917 @@
+                            f"canonical magnitude {field} must be finite"
+                        )
+                if (
+                    _decimal(value["source_unit_usd"]) <= 0
+                    or _decimal(value["conversion_factor"]) <= 0
+                ):
+                    raise RewardOntologyError("canonical magnitude scale must be positive")
+                expected_factor = (
+                    _decimal(value["source_unit_usd"]) / CANONICAL_UNIT_USD
+                )
+                if _decimal(value["conversion_factor"]) != expected_factor:
+                    raise RewardOntologyError("canonical conversion factor mismatch")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)